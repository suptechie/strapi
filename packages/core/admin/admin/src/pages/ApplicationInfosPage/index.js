// import React from 'react';
// import { Header } from '@buffetjs/custom';
// import { Flex, Padded, Text } from '@buffetjs/core';
// import { useIntl } from 'react-intl';
// import { BaselineAlignment, useAppInfos } from '@strapi/helper-plugin';
// import Bloc from '../../components/Bloc';
// import PageTitle from '../../components/SettingsPageTitle';
// import { Detail, InfoText } from './components';

// const ApplicationInfosPage = () => {
//   const appInfos = useAppInfos();
//   const { shouldUpdateStrapi, latestStrapiReleaseTag } = appInfos;
//   const { formatMessage } = useIntl();

//   const currentPlan = appInfos.communityEdition
//     ? 'app.components.UpgradePlanModal.text-ce'
//     : 'app.components.UpgradePlanModal.text-ee';

//   const headerProps = {
//     title: { label: formatMessage({ id: 'Settings.application.title' }) },
//     content: formatMessage({
//       id: 'Settings.application.description',
//     }),
//   };
//   const pricingLabel = formatMessage({ id: 'Settings.application.link-pricing' });
//   const upgradeLabel = formatMessage({ id: 'Settings.application.link-upgrade' });
//   const strapiVersion = formatMessage({ id: 'Settings.application.strapi-version' });
//   const nodeVersion = formatMessage({ id: 'Settings.application.node-version' });
//   const editionTitle = formatMessage({ id: 'Settings.application.edition-title' });

//   /* eslint-disable indent */
//   const upgradeLink = shouldUpdateStrapi
//     ? {
//         label: upgradeLabel,
//         href: `https://github.com/strapi/strapi/releases/tag/${latestStrapiReleaseTag}`,
//       }
//     : null;
//   /* eslint-enable indent */

//   return (
//     <div>
//       <PageTitle name="Application" />
//       <Header {...headerProps} />
//       <BaselineAlignment top size="3px" />
//       <Bloc>
//         <Padded left right top size="smd">
//           <Padded left right top size="xs">
//             <Flex justifyContent="space-between">
//               <Detail
//                 link={upgradeLink}
//                 title={strapiVersion}
//                 content={`v${appInfos.strapiVersion}`}
//               />
//               <Detail
//                 link={{ label: pricingLabel, href: 'https://strapi.io/pricing' }}
//                 title={editionTitle}
//                 content={formatMessage({ id: currentPlan })}
//               />
//             </Flex>
//             <Padded top size="lg">
//               <Text fontSize="xs" color="grey" fontWeight="bold">
//                 {nodeVersion}
//               </Text>
//               <InfoText content={appInfos.nodeVersion} />
//             </Padded>
//           </Padded>
//         </Padded>
//         <BaselineAlignment top size="60px" />
//       </Bloc>
//     </div>
//   );
// };

// export default ApplicationInfosPage;
export default () => 'ApplicationInfosPage';
