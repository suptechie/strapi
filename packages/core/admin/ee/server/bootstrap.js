'use strict';

const _ = require('lodash');
// eslint-disable-next-line node/no-extraneous-require
const { features } = require('@strapi/strapi/lib/utils/ee');
const executeCEBootstrap = require('../../server/bootstrap');
const { getService } = require('../../server/utils');
const actions = require('./config/admin-actions');

const enableUsersToLicenseLimit = async (numberOfUsersToEnable) => {
  const data = await strapi.db.query('strapi::ee-store').findOne({
    where: { key: 'ee_disabled_users' },
  });

  if (!data || !data.value || data.value.length === 0) return;

  const disabledUsers = JSON.parse(data.value);

  const usersToEnable = _.take(disabledUsers, numberOfUsersToEnable);

  usersToEnable.forEach(async (user) => {
    const data = await strapi.db.query('admin::user').findOne({
      where: { id: user.id },
    });

    if (!data) return;

    await strapi.db.query('admin::user').update({
      where: { id: user.id },
      data: { isActive: true },
    });
  });

  const remainingDisabledUsers = _.drop(disabledUsers, numberOfUsersToEnable);

  return strapi.db.query('strapi::ee-store').update({
    where: { id: data.id },
    data: { value: JSON.stringify(remainingDisabledUsers) },
  });
};

const calculateAdminSeatDifference = async (seatsAllowedByLicense) => {
  const currentAdminSeats = await strapi.db.query('admin::user').count({
    where: {
      isActive: true,
    },
  });
  return currentAdminSeats - seatsAllowedByLicense;
};

const disableUsersAboveLicenseLimit = async (numberOfUsersToDisable) => {
  const users = await strapi.db.query('admin::user').findMany({
    where: { isActive: 'true' },
    orderBy: { createdAt: 'DESC' },
    populate: { roles: { select: ['id'] } },
  });

  const usersToDisable = _.take(users, numberOfUsersToDisable);

  usersToDisable.forEach(async (user) => {
    user.isActive = false;
    await strapi.db.query('admin::user').update({
      where: { id: user.id },
      data: {
        isActive: false,
      },
    });
  });

  const data = await strapi.db.query('strapi::ee-store').findOne({
    where: { key: 'ee_disabled_users' },
  });

  if (data) {
    return strapi.db.query('strapi::ee-store').update({
      where: { id: data.id },
      data: { value: JSON.stringify(usersToDisable) },
    });
  }

  return strapi.db.query('strapi::ee-store').create({
    data: {
      key: 'ee_disabled_users',
      value: JSON.stringify(usersToDisable),
    },
  });
};

const syncdDisabledUserRecords = async () => {
  const data = await strapi.db.query('strapi::ee-store').findOne({
    where: { key: 'ee_disabled_users' },
  });

  if (!data || !data.value || data.value.length === 0) return;

  const disabledUsers = JSON.parse(data.value);
  disabledUsers.forEach(async (user) => {
    const data = await strapi.db.query('admin::user').findOne({
      where: { id: user.id },
    });

    if (!data) return;

    return strapi.db.query('admin::user').update({
      where: { id: user.id },
      data: { isActive: user.isActive }, // TODO: should this value be hardcoded to 'false' or no?
    });
  });
};

module.exports = async () => {
  const { actionProvider } = getService('permission');

  if (features.isEnabled('sso')) {
    await actionProvider.registerMany(actions.sso);
  }

  if (features.isEnabled('audit-logs')) {
    await actionProvider.registerMany(actions.auditLogs);
  }

  // TODO: check admin seats
  await syncdDisabledUserRecords();

  const permittedAdminSeats = 5;

  const adminSeatDifference = await calculateAdminSeatDifference(permittedAdminSeats);

  switch (true) {
    case adminSeatDifference === 0:
      break;
    case adminSeatDifference > 0:
      await disableUsersAboveLicenseLimit(adminSeatDifference);
      break;
    case adminSeatDifference < 0:
      await enableUsersToLicenseLimit(Math.abs(adminSeatDifference));
      break;
    default:
      break;
  }
  await executeCEBootstrap();
};
