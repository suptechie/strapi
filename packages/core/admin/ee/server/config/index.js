'use strict';

module.exports = {
  functions: {
    bootstrap: require('./functions/bootstrap'),
  },
  routes: require('./routes').routes,
};
