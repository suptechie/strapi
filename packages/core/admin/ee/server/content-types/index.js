'use strict';

const workflow = require('./workflow');

module.exports = {
  workflow,
};
