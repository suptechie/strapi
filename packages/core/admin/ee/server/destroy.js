'use strict';

const { features } = require('@strapi/strapi/dist/utils/ee');
const executeCEDestroy = require('../../server/destroy');

module.exports = async ({ strapi }) => {
  if (features.isEnabled('audit-logs')) {
    strapi.get('audit-logs').destroy();
  }

  await executeCEDestroy();
};
