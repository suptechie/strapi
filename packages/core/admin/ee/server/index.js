'use strict';

module.exports = {
  register: require('./register'),
  bootstrap: require('./bootstrap'),
  routes: require('./routes'),
  services: require('./services'),
  controllers: require('./controllers'),
};
