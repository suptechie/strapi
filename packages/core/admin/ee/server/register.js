'use strict';

const executeCERegister = require('../../server/register');

module.exports = async ({ strapi }) => {
  // TODO: register auditLogs provider here
  await executeCERegister({ strapi });
};
