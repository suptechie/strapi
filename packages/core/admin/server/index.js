'use strict';

const { getService } = require('./utils');

module.exports = {
  // TODO: update load middleware to not load the admin middleware from here
  // TODO: load bootstrap / register independently
  async destroy() {
    const { conditionProvider, actionProvider } = getService('permission');

    await conditionProvider.clear();
    await actionProvider.clear();
  },
  config: require('./config'),
  services: require('./services'),
  controllers: require('./controllers'),
  models: require('./content-types'),
};
