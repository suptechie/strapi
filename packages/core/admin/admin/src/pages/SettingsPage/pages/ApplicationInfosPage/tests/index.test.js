import React from 'react';

import { lightTheme, ThemeProvider } from '@strapi/design-system';
import { TrackingProvider, useAppInfo, useRBAC } from '@strapi/helper-plugin';
import { fireEvent, render, screen, waitFor } from '@testing-library/react';
import { IntlProvider } from 'react-intl';
import { QueryClient, QueryClientProvider } from 'react-query';

import ApplicationInfosPage from '../index';

import server from './server';

const updateProjectSettingsSpy = jest.fn();

jest.mock('@strapi/helper-plugin', () => ({
  ...jest.requireActual('@strapi/helper-plugin'),
  // eslint-disable-next-line
  CheckPermissions: ({ children }) => <div>{children}</div>,
  useAppInfo: jest.fn(() => ({ shouldUpdateStrapi: false, latestStrapiReleaseTag: 'v3.6.8' })),
  useNotification: jest.fn(() => jest.fn()),
  useRBAC: jest.fn(() => ({ allowedActions: { canRead: true, canUpdate: true } })),
  useFetchClient: jest.fn().mockReturnValue({
    get: jest.fn().mockResolvedValue({
      data: {
        menuLogo: {
          ext: 'png',
          height: 256,
          name: 'image.png',
          size: 27.4,
          url: 'uploads/image_fe95c5abb9.png',
          width: 246,
        },
      },
    }),
  }),
}));
jest.mock('../../../../../hooks', () => ({
  useConfigurations: jest.fn(() => ({
    logos: {
      menu: { custom: 'customMenuLogo.png', default: 'defaultMenuLogo.png' },
      auth: { custom: 'customAuthLogo.png', default: 'defaultAuthLogo.png' },
    },
    updateProjectSettings: updateProjectSettingsSpy,
  })),
}));
jest.mock(
  'ee_else_ce/pages/SettingsPage/pages/ApplicationInfosPage/components/AdminSeatInfo',
  () => () => {
    return <></>;
  }
);

const client = new QueryClient();

const App = (
  <QueryClientProvider client={client}>
    <TrackingProvider>
      <ThemeProvider theme={lightTheme}>
        <IntlProvider locale="en" messages={{}} textComponent="span">
          <ApplicationInfosPage />
        </IntlProvider>
      </ThemeProvider>
    </TrackingProvider>
  </QueryClientProvider>
);

describe('Application page', () => {
  beforeAll(() => server.listen());

  afterEach(() => {
    server.resetHandlers();
    jest.restoreAllMocks();
  });

  afterAll(() => server.close());

  it('should not display link upgrade version if not necessary', () => {
    const { queryByText } = render(App);

    expect(queryByText('Upgrade your admin panel')).not.toBeInTheDocument();
  });

  it('should display upgrade version warning if the version is behind the latest one', () => {
    useAppInfo.mockImplementationOnce(() => {
      return {
        shouldUpdateStrapi: true,
        latestStrapiReleaseTag: 'v3.6.8',
        strapiVersion: '4.0.0',
      };
    });

    render(App);

    expect(screen.getByText('v4.0.0')).toBeInTheDocument();
    expect(screen.getByText('Upgrade your admin panel')).toBeInTheDocument();
  });

  it('should render logo input if read permissions', async () => {
    const { queryByText } = render(App);

    await waitFor(() => {
      expect(queryByText('Menu logo')).toBeInTheDocument();
    });
  });

  it('should not render logo input if no read permissions', async () => {
    useRBAC.mockImplementationOnce(() => ({
      allowedActions: { canRead: false, canUpdate: false },
    }));
    const { queryByText } = render(App);

    await waitFor(() => {
      expect(queryByText('Menu logo')).not.toBeInTheDocument();
    });
  });

  it('should render save button if update permissions', async () => {
    const { queryByText } = render(App);

    await waitFor(() => {
      expect(queryByText('Save')).toBeInTheDocument();
    });
  });

  it('should not render save button if no update permissions', async () => {
    useRBAC.mockImplementationOnce(() => ({ allowedActions: { canRead: true, canUpdate: false } }));
    const { queryByText } = render(App);

    await waitFor(() => {
      expect(queryByText('Save')).not.toBeInTheDocument();
    });
  });

  it('should update project settings on save', async () => {
    const { getByRole } = render(App);

    fireEvent.click(getByRole('button', { name: 'Save' }));
    await waitFor(() => expect(updateProjectSettingsSpy).toHaveBeenCalledTimes(1));
  });
});
