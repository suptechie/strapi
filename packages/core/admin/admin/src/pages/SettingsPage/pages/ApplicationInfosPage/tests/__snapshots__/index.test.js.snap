// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Application page renders and matches the snapshot 1`] = `
.c2 {
  padding-bottom: 56px;
}

.c5 {
  background: #f6f6f9;
  padding-top: 40px;
  padding-right: 56px;
  padding-bottom: 40px;
  padding-left: 56px;
}

.c12 {
  padding-right: 8px;
}

.c16 {
  padding-right: 56px;
  padding-left: 56px;
}

.c19 {
  background: #ffffff;
  padding-top: 24px;
  padding-right: 32px;
  padding-bottom: 24px;
  padding-left: 32px;
  border-radius: 4px;
  box-shadow: 0px 1px 4px rgba(33,33,52,0.1);
}

.c22 {
  padding-top: 4px;
}

.c30 {
  padding-left: 8px;
}

.c6 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c7 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
}

.c17 {
  -webkit-align-items: stretch;
  -webkit-box-align: stretch;
  -ms-flex-align: stretch;
  align-items: stretch;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c9 {
  font-weight: 600;
  font-size: 2rem;
  line-height: 1.25;
  color: #32324d;
}

.c14 {
  font-size: 0.75rem;
  line-height: 1.33;
  font-weight: 600;
  line-height: 1.14;
  color: #32324d;
}

.c15 {
  font-size: 1rem;
  line-height: 1.5;
  color: #666687;
}

.c21 {
  font-weight: 500;
  font-size: 1rem;
  line-height: 1.25;
  color: #32324d;
}

.c25 {
  font-weight: 600;
  font-size: 0.6875rem;
  line-height: 1.45;
  text-transform: uppercase;
  color: #666687;
}

.c26 {
  font-size: 0.875rem;
  line-height: 1.43;
  color: #32324d;
}

.c29 {
  font-size: 0.875rem;
  line-height: 1.43;
  color: #4945ff;
}

.c18 > * {
  margin-top: 0;
  margin-bottom: 0;
}

.c18 > * + * {
  margin-top: 24px;
}

.c20 > * {
  margin-top: 0;
  margin-bottom: 0;
}

.c20 > * + * {
  margin-top: 20px;
}

.c10 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  cursor: pointer;
  padding: 8px;
  border-radius: 4px;
  background: #ffffff;
  border: 1px solid #dcdce4;
  position: relative;
  outline: none;
}

.c10 svg {
  height: 12px;
  width: 12px;
}

.c10 svg > g,
.c10 svg path {
  fill: #ffffff;
}

.c10[aria-disabled='true'] {
  pointer-events: none;
}

.c10:after {
  -webkit-transition-property: all;
  transition-property: all;
  -webkit-transition-duration: 0.2s;
  transition-duration: 0.2s;
  border-radius: 8px;
  content: '';
  position: absolute;
  top: -4px;
  bottom: -4px;
  left: -4px;
  right: -4px;
  border: 2px solid transparent;
}

.c10:focus-visible {
  outline: none;
}

.c10:focus-visible:after {
  border-radius: 8px;
  content: '';
  position: absolute;
  top: -5px;
  bottom: -5px;
  left: -5px;
  right: -5px;
  border: 2px solid #4945ff;
}

.c13 {
  height: 100%;
}

.c11 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #4945ff;
  border: 1px solid #4945ff;
  height: 2rem;
  padding-left: 16px;
  padding-right: 16px;
}

.c11 .c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c11 .c8 {
  color: #ffffff;
}

.c11[aria-disabled='true'] {
  border: 1px solid #dcdce4;
  background: #eaeaef;
}

.c11[aria-disabled='true'] .c8 {
  color: #666687;
}

.c11[aria-disabled='true'] svg > g,.c11[aria-disabled='true'] svg path {
  fill: #666687;
}

.c11[aria-disabled='true']:active {
  border: 1px solid #dcdce4;
  background: #eaeaef;
}

.c11[aria-disabled='true']:active .c8 {
  color: #666687;
}

.c11[aria-disabled='true']:active svg > g,.c11[aria-disabled='true']:active svg path {
  fill: #666687;
}

.c11:hover {
  border: 1px solid #7b79ff;
  background: #7b79ff;
}

.c11:active {
  border: 1px solid #4945ff;
  background: #4945ff;
}

.c11 svg > g,
.c11 svg path {
  fill: #ffffff;
}

.c27 {
  cursor: pointer;
}

.c28 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  position: relative;
  outline: none;
}

.c28 svg path {
  fill: #4945ff;
}

.c28 svg {
  font-size: 0.625rem;
}

.c28:after {
  -webkit-transition-property: all;
  transition-property: all;
  -webkit-transition-duration: 0.2s;
  transition-duration: 0.2s;
  border-radius: 8px;
  content: '';
  position: absolute;
  top: -4px;
  bottom: -4px;
  left: -4px;
  right: -4px;
  border: 2px solid transparent;
}

.c28:focus-visible {
  outline: none;
}

.c28:focus-visible:after {
  border-radius: 8px;
  content: '';
  position: absolute;
  top: -5px;
  bottom: -5px;
  left: -5px;
  right: -5px;
  border: 2px solid #4945ff;
}

.c31 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c1 {
  display: grid;
  grid-template-columns: 1fr;
}

.c3 {
  overflow-x: hidden;
}

.c23 {
  display: grid;
  grid-template-columns: repeat(12,1fr);
  gap: 0px;
}

.c24 {
  grid-column: span 6;
  max-width: 100%;
}

.c4:focus-visible {
  outline: none;
}

@media (max-width:68.75rem) {
  .c24 {
    grid-column: span 12;
  }
}

@media (max-width:34.375rem) {
  .c24 {
    grid-column: span;
  }
}

<div
  class="c0 c1"
>
  <div
    class="c0 c2 c3"
  >
    <main
      aria-labelledby="main-content-title"
      class="c4"
      id="main-content"
      tabindex="-1"
    >
      <form>
        <div
          style="height: 0px;"
        >
          <div
            class="c0 c5"
            data-strapi-header="true"
          >
            <div
              class="c0 c6"
            >
              <div
                class="c0 c7"
              >
                <h1
                  class="c8 c9"
                >
                  Overview
                </h1>
              </div>
              <button
                aria-disabled="false"
                class="c10 c11"
                type="submit"
              >
                <div
                  aria-hidden="true"
                  class="c0 c12 c13"
                >
                  <svg
                    fill="none"
                    height="1em"
                    viewBox="0 0 24 24"
                    width="1em"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M20.727 2.97a.2.2 0 01.286 0l2.85 2.89a.2.2 0 010 .28L9.554 20.854a.2.2 0 01-.285 0l-9.13-9.243a.2.2 0 010-.281l2.85-2.892a.2.2 0 01.284 0l6.14 6.209L20.726 2.97z"
                      fill="#212134"
                    />
                  </svg>
                </div>
                <span
                  class="c8 c14"
                >
                  Save
                </span>
              </button>
            </div>
            <p
              class="c8 c15"
            >
              Administration panelâ€™s global information
            </p>
          </div>
        </div>
        <div
          class="c0 c16"
        >
          <div
            class="c0 c17 c18"
            spacing="6"
          >
            <div
              class="c0 c19 c17 c20"
              spacing="5"
            >
              <h3
                class="c8 c21"
              >
                Details
              </h3>
              <div
                class="c0 c22 c23"
              >
                <div
                  class="c24"
                >
                  <div
                    class="c0 "
                  >
                    <span
                      class="c8 c25"
                    >
                      strapi version
                    </span>
                    <p
                      class="c8 c26"
                    >
                      v
                    </p>
                    <a
                      class="c27 c28"
                      href="https://support.strapi.io/support/home"
                      rel="noreferrer noopener"
                      target="_blank"
                    >
                      <span
                        class="c8 c29"
                      >
                        Get help
                      </span>
                      <span
                        aria-hidden="true"
                        class="c0 c30 c31"
                      >
                        <svg
                          fill="none"
                          height="1em"
                          viewBox="0 0 24 24"
                          width="1em"
                          xmlns="http://www.w3.org/2000/svg"
                        >
                          <path
                            d="M16.235 2.824a1.412 1.412 0 010-2.824h6.353C23.368 0 24 .633 24 1.412v6.353a1.412 1.412 0 01-2.823 0V4.82l-8.179 8.178a1.412 1.412 0 01-1.996-1.996l8.178-8.178h-2.945zm4.942 10.588a1.412 1.412 0 012.823 0v9.176c0 .78-.632 1.412-1.412 1.412H1.412C.632 24 0 23.368 0 22.588V1.412C0 .632.632 0 1.412 0h9.176a1.412 1.412 0 010 2.824H2.824v18.353h18.353v-7.765z"
                            fill="#32324D"
                          />
                        </svg>
                      </span>
                    </a>
                  </div>
                </div>
                <div
                  class="c24"
                >
                  <div
                    class="c0 "
                  >
                    <span
                      class="c8 c25"
                    >
                      current plan
                    </span>
                    <p
                      class="c8 c26"
                    >
                      Enterprise Edition
                    </p>
                  </div>
                </div>
              </div>
              <div
                class="c0 c22 c23"
              >
                <div
                  class="c24"
                >
                  <div
                    class="c0 "
                  />
                </div>
                <div
                  class="c24"
                >
                  <div
                    class="c0 "
                  >
                    <a
                      class="c27 c28"
                      href="https://strapi.io/pricing-self-hosted"
                      rel="noreferrer noopener"
                      target="_blank"
                    >
                      <span
                        class="c8 c29"
                      >
                        See all pricing plans
                      </span>
                      <span
                        aria-hidden="true"
                        class="c0 c30 c31"
                      >
                        <svg
                          fill="none"
                          height="1em"
                          viewBox="0 0 24 24"
                          width="1em"
                          xmlns="http://www.w3.org/2000/svg"
                        >
                          <path
                            d="M16.235 2.824a1.412 1.412 0 010-2.824h6.353C23.368 0 24 .633 24 1.412v6.353a1.412 1.412 0 01-2.823 0V4.82l-8.179 8.178a1.412 1.412 0 01-1.996-1.996l8.178-8.178h-2.945zm4.942 10.588a1.412 1.412 0 012.823 0v9.176c0 .78-.632 1.412-1.412 1.412H1.412C.632 24 0 23.368 0 22.588V1.412C0 .632.632 0 1.412 0h9.176a1.412 1.412 0 010 2.824H2.824v18.353h18.353v-7.765z"
                            fill="#32324D"
                          />
                        </svg>
                      </span>
                    </a>
                  </div>
                </div>
              </div>
              <div
                class="c0 c22"
              >
                <span
                  class="c8 c25"
                >
                  node version
                </span>
                <p
                  class="c8 c26"
                />
              </div>
            </div>
          </div>
        </div>
      </form>
    </main>
  </div>
</div>
`;
